### PR TITLE
[DEV-477] XML Parsing Limits

### PR DESCRIPTION
**Pull request by e45lee**
Limits taken from https://docs.oracle.com/javase/tutorial/jaxp/limits/using.html -- locally tested, doesn't break any existing SAML test files we have.

### CODE REVIEW

File: test-resources/xml-bomb.txt
Comment (Elknar) [inline test-resources/xml-bomb.txt:1]: please change the file extension to align with the contents

Comment (e45lee) [reply test-resources/xml-bomb.txt:1]: I left it as `.txt` as leaving it as `.xml` caused some unpleasant surprises when I tried to edit it (my editor froze up)

File: test/saml20_clj/sp_test.clj
Comment (Elknar) [inline test/saml20_clj/sp_test.clj]: Minor issue, but when the test case fails (e.g. if the size restriction is not honored), the test will throw an NPE error (and will be marked as an Error rather than Fail)

Imho, it would be better to test the `saml20-clj.xml/str->xmldoc` method itself, rather than `xml-string->saml-resp`, as the former does not include SAML-related logic and is only about XML processing

**Review by Elknar — CHANGES_REQUESTED** [review]

File: test-resources/xml-bomb.txt
Comment (Elknar) [reply test-resources/xml-bomb.txt:1]: oh... uh... yeah, that can happen. nvm then

File: test/saml20_clj/sp_test.clj
Comment (e45lee) [reply test/saml20_clj/sp_test.clj]: The NPE is annoying but a small thing -- I've changed this to test saml20-clj.xml/str->xmldoc

File: test/saml20_clj/sp_test.clj
Comment (Elknar) [inline test/saml20_clj/sp_test.clj]: This may no longer be the correct namespace for this test then :(

**Review by Elknar — CHANGES_REQUESTED** [review]
also: `java.lang.RuntimeException: Unmatched delimiter: )`

File: test/saml20_clj/sp_test.clj
Comment (Elknar) [inline test/saml20_clj/sp_test.clj]: `Exception in thread "main" java.lang.Exception: :only/:refer value must be a sequential collection of symbols, compiling:(saml20_clj/sp_test.clj:1:1)`

missing `:` before all.

meanwhile, i still suggest moving this to a separate ns

**Review by Elknar — CHANGES_REQUESTED** [review]

**Review by Elknar — APPROVED** [review]
Need second review I guess

File: src/saml20_clj/xml.clj
Comment (MatthewDarling) [inline src/saml20_clj/xml.clj:35]: The [example page](https://docs.oracle.com/javase/tutorial/jaxp/limits/using.html) uses `PARAMETER_ENTITY_SIZE_LIMIT`, without "max".

I'm not sure which is right, because Oracle is too cheap to host something useful like documentation at that URL.

The rest all seem to line up, at least :)

Comment (e45lee) [reply src/saml20_clj/xml.clj:35]: I double checked and it probably matches up: https://docs.oracle.com/javase/tutorial/jaxp/limits/limits.html

Comment (MatthewDarling) [reply src/saml20_clj/xml.clj:35]: Ahhh, that's the page I was looking for. Thanks!

Yep, looks good.

**Review by MatthewDarling — APPROVED** [review]

**Comment by e45lee** [comment]
Thanks!

**Pull request merged**
